docs/architecture/dapr-for-net-developers/actors.md

```diff
@@ -0,0 +1,636 @@
+    }
+
+    return newValue;
+}
+```
+
+Unfortunately, using explicit locking mechanisms is error-prone. They can easily lead to deadlocks and can have serious impact on performance.
+
+Thanks to the turn-based access model, you don't need to worry about multiple threads with actors, making it much easier to concurrent systems. The following actor example closely mirrors the code from the previous sample, but doesn't require any locking mechanisms to be correct:
+
+```csharp
+public async Task<int> IncrementAsync()
+{
+    var counterValue = await StateManager.TryGetStateAsync<int>("counter");
+
+    var currentValue = counterValue.HasValue ? counterValue.Value : 0;
+    var newValue = currentValue + 1;
+
+    await StateManager.SetStateAsync("counter", newValue);
+
+    return newValue;
+}
+```
+
+### Timers and reminders
+
+Actors can use timers and reminders to schedule calls to themselves. Both concepts support the configuration of a due time. The difference lies in the lifetime of the callback registrations:
+
+- Timers will only stay active as long as the the actor is activated. Timers *will not* reset the idle-timer, so they cannot keep an actor active on their own.
+- Reminders outlive actor activations. If an actor is deactivated, a reminder will re-activate the actor. Reminders *will* reset the idle-timer.
+
+Timers are registered by making a call to the actor API. In the following example, a timer is registered with a due time of 0 and a period of 10 seconds.
+
+```bash
+curl -X POST http://localhost:3500/v1.0/actors/<actorType>/<actorId>/timers/<name> \
+  -H "Content-Type: application/json" \
+  -d '{
+        "dueTime": "0h0m0s0ms",
+        "period": "0h0m10s0ms"
+      }'
+```
+
+Because the due time is 0, the timer will fire immediately. After a timer callback has finished, the timer will wait 10 seconds before firing again.
+
+Reminders are registered in a similar way. The following example shows a reminder registration with a due time of 5 minutes, and an empty period:
+
+```bash
+curl -X POST http://localhost:3500/v1.0/actors/<actorType>/<actorId>/reminders/<name> \
+  -H "Content-Type: application/json" \
+  -d '{
+        "dueTime": "0h5m0s0ms",
+        "period": ""
+      }'
+```
+
+This reminder will fire in 5 minutes. Because the given period is empty, this will be a one-time reminder.
+
+> [!NOTE]
+> Timers and reminders both respect the turn-based access model. When a timer or reminder fires, the callback will not be executed until any other method invocation or timer/reminder callback has finished.
+
+### State persistence
+
+Actor state is persisted using the Dapr [state management building block](state-management.md). Because actors can execute multiple state operations in a single turn, the state store component must support multi-item transactions. At the time of writing, the following state stores support multi-item transactions:
+
+- Azure Cosmos DB
+- MongoDB
+- PostgreSQL
+- Redis
+- SQL Server
```

Comment:
There are 7 in total as of today. See https://docs.dapr.io/reference/components-reference/supported-state-stores/